### DISCUSSION TITLE
why we need a plugin for every static binary tool？

### DISCUSSION THREAD

**Discussion by dfang** [jdx/mise#686] [Ideas]
for static binary tools that written by go or rust, like jq, fd, fzf, curlie, shfmt,  the installation method shares a pattern: you download it from GitHub release and extract to a PATH, chmod +x, then you can execute it.  normally we don't need multi versions coexist on our system, just an executable in PATH is enough. so we don't need a plugin.  even if it's necessary, one plugin is for all.

@jdxcode

for inspiration
- https://gobinaries.com
- https://webinstall.dev/
- https://tea.xyz

binaries
- https://github.com/sharkdp/fd
- https://github.com/junegunn/fzf
- https://github.com/jqlang/jq/
- https://github.com/ajeetdsouza/zoxide
- https://github.com/rs/curlie
- https://github.com/httpie/httpie
- https://github.com/patrickvane/shfmt
- https://cli.github.com

**Comment by jdx** [comment]
just an FYI: this is something I want to do. I'm thinking possibly in February I could get around to it. Right now I'm mostly focused on finishing up tasks.

**Reply by shah** [reply]
@jdx if you integrate [`pkgx`](https://pkgx.sh/) (the new name for `tea`, by the author of Brew I think) and [`libpkgx`](https://github.com/pkgxdev/libpkgx) you should be able to get all those features plus if you integrate [`eget`](https://github.com/zyedidia/eget) you'll get all GitHub binaries basically "for free".

In some respects, `pkgx` and `mise` are competitors but I think they could be quite complementary as well. `eget` is definitely complementary.

**Reply by jdx** [reply]
what I'd really like to see is plugins being capable of that integration so there could just be an eget/pkgx/go get plugin to install other tools

**Reply by shah** [reply]
Great idea @jdx!

**Comment by jdx** [comment]
closing in favor of https://github.com/jdx/mise/discussions/1250
